app/client-v2/src/components/ProjectView/ProjectPage.vue

```diff
@@ -3,12 +3,17 @@ import { useProjectStore } from "@/stores/projectStore";
 import { useRoute } from "vue-router";
 import ProjectPostRun from "@/components/ProjectView/ProjectPostRun.vue";
 import ProjectPreRun from "@/components/ProjectView/ProjectPreRun.vue";
-import { onUnmounted } from "vue";
+import { ref, onMounted, onUnmounted } from "vue";
 import Toast from "primevue/toast";
 
 const route = useRoute();
 const store = useProjectStore();
-const projectFetchError = await store.getProject(route.params.id as string);
+
+let projectFetchError = ref<any>(null);
+
+onMounted(async () => {
```

Comment:
sorry why is this being done on mount? we ideally want to fetch data before anything has rendered

Reply:
See below.